filament/src/details/Texture.cpp

```diff
@@ -242,18 +242,30 @@ Texture* Texture::Builder::build(Engine& engine) {
 
     if (mImpl->mUsage == TextureUsage::NONE) {
         mImpl->mUsage = TextureUsage::DEFAULT;
-        if (mImpl->mLevels > 1 &&
-            (mImpl->mWidth > 1 || mImpl->mHeight > 1) &&
-            !mImpl->mExternal) {
-            const bool formatMipmappable =
-                    downcast(engine).getDriverApi().isTextureFormatMipmappable(mImpl->mFormat);
-            if (formatMipmappable) {
-                // by default mipmappable textures have the BLIT usage bits set
-                mImpl->mUsage |= TextureUsage::BLIT_SRC | TextureUsage::BLIT_DST;
-            }
+    }
+
+    auto const& featureFlags = downcast(engine).features.engine.debug;
+    if (mImpl->mLevels > 1 && (mImpl->mWidth > 1 || mImpl->mHeight > 1) && !mImpl->mExternal) {
+        bool const formatMipmappable =
+                downcast(engine).getDriverApi().isTextureFormatMipmappable(mImpl->mFormat);
+
+        FILAMENT_FLAG_GUARDED_CHECK_PRECONDITION(
+                formatMipmappable,
+                featureFlags.assert_texture_format_mipmappable)
+                << "Texture levels is > 1, but the format (" << int(mImpl->mFormat) << ") "
+                << " is not mipmppable";
+
+        if (formatMipmappable && !featureFlags.assert_texture_can_generate_mipmap) {
+            // Setting this for backwards compatibility,but should remove when safe.
+            mImpl->mUsage |= TextureUsage::GEN_MIPMAPPABLE;
         }
     }
 
+    if (any(mImpl->mUsage & TextureUsage::GEN_MIPMAPPABLE)) {
+        FILAMENT_CHECK_PRECONDITION(mImpl->mLevels > 1)
+                << "Texture has GEN_MIPMAPPABLE usage set, but levels is 1.";
+    }
```

Comment:
I think we shouldn't assert here, it's dumb to do this, but legal.

Reply:
removed